automata/fa/dfa.py

```diff
@@ -509,34 +480,15 @@ def intersection_function(state_pair: Tuple[DFAStateT, DFAStateT]) -> bool:
             q_a, q_b = state_pair
             return q_a in self.final_states and q_b in other.final_states
 
-        (
-            new_states,
-            new_transitions,
-            new_initial_state,
-            new_final_states,
-        ) = self._cross_product(
-            other,
+        initial_state, expand_state_fn = DFA._cross_product(self, other)
+
+        return DFA._expand_dfa(
```

Comment:
I think some of these should be `self.__class__` to play nicer with inheritance? Double check with @caleb531

Reply:
@EduardoGoulart1 Yes, I generally prefer `self.__class__` (except under rare circumstances) when returning new instances within a class.

Reply:
I see the code has been updated to use `self.__class__`, so resolving this conversation thread now.